refactory: move build

Juanperias opened the pull request:
This refactoring changes the name of the variable dest_dir and also moves the contents of the build command to commands

Juanperias committed:
refactory: Change var names and move  build

This refactoring changes the name of the variable dest_dir and also moves the contents of the build command to commands

SergioRibera commented on src/commands.rs:
```diff
@@ -33,7 +33,7 @@ pub enum Commands {
         /// If omitted, mdBook uses build.build-dir from book.toml \
         /// or defaults to `./book`.
         #[clap(long, short, value_hint = ValueHint::DirPath)]
-        dest_dir: Option<PathBuf>,
```
This breaks with our idea of backward compatibility

https://rust-lang.github.io/mdBook/cli/build.html#--dest-dir

SergioRibera reviewed (changes requested):
The rest I think is great, I have only this to say

Juanperias committed:
Refactor: Move build

This refactoring moves to the clean function and fixes a problem that happened in the other refactoring.

SergioRibera reviewed (approved):
Thank you very much for your contribution
